src/main.rs

```diff
@@ -1,8 +1,60 @@
 use geo::Point;
 use serde_json::to_string_pretty;
 use zonebuilder::{clockboard, Params};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// A basic example
+#[derive(StructOpt, Debug)]
+#[structopt(name = "zb")]
+struct Opt {
+    // A flag, true if used in the command line. Note doc comment will
+    // be used for the help message of the flag. The name of the
+    // argument will be, by default, based on the name of the field.
+    /// Activate debug mode
+    #[structopt(short, long)]
+    debug: bool,
+
+    // The number of occurrences of the `v/verbose` flag
+    /// Verbose mode (-v, -vv, -vvv, etc.)
+    #[structopt(short, long, parse(from_occurrences))]
+    verbose: u8,
+
+    /// Set n_circles
+    #[structopt(short, long, default_value = "5")]
+    n_circles: usize,
+
+    /// Number of radial segments (12 by default)
+    #[structopt(short, long, default_value = "12")]
+    num_segments: usize,
+
+    /// Distances between concentric rings.
+    /// First 5 values of the triangular number sequence (1 to 15)
+    /// by default
+    #[structopt(short, long, default_value = "vec![1.0, 3.0, 6.0, 10.0, 15.0]")]
+    distances: Vec<f64>,
+
+    /// Number of vertices per arc
+    #[structopt(short, long, default_value = "5")]
+    num_vertices_arc: usize,
+
+    /// Number of decimal places in the resulting output (GeoJSON) files.
+    /// Set to 6 by default. Larger numbers mean more precision but
+    /// larger file sizes.
+    #[structopt(short, long, default_value = "5")]
+    precision: usize,
+
+    /// Output file
+    #[structopt(short, long, parse(from_os_str))]
+    output: PathBuf,
+
+}
 
 fn main() {
+
+    let opt = Opt::from_args();
```

Comment:
This is a start, but the options don't get used anywhere. You could either populate `Params` (from `lib.rs`) with these options, or... more directly, just derive structopt on the existing `Params`.

Reply:
> or... more directly, just derive structopt on the existing `Params`.

Could you explain? I tried this:

```
    #[structopt(short = "c", long, default_value = params.n_circles)]
```

(bit of a stab in the dark)

Reply:
I mean get rid of `Opt` and go make all of the changes on the `Params` struct instead. Then the command-line values will get used. I don't see a good reason why both `Opt` and `Params` should exist right now.

Reply:
Aha, yes and the default value system works well also. So will aim to rename Opt as Params and delete the original Params. Yes there is duplication that requires some kind of refactor. More soon!